Forwarded: http://anonsvn.wireshark.org/viewvc?view=rev&revision=34080

Backported from upstream:
r34080 | stig | 2010-09-08 15:38:47 +0200 (Wed, 08 Sep 2010) | 3 lines

Revert revision 33780, because the checksum is only calculated
from the header and not the content.

Index: epan/dissectors/packet-ip.c
===================================================================
--- a/epan/dissectors/packet-ip.c	(revision 34079)
+++ b/epan/dissectors/packet-ip.c	(revision 34080)
@@ -1502,7 +1502,7 @@
   /*
    * If we have the entire IP header available, check the checksum.
    */
-  if (ip_check_checksum && tvb_bytes_exist(tvb, offset, hlen) && !pinfo->in_error_pkt) {
+  if (ip_check_checksum && tvb_bytes_exist(tvb, offset, hlen)) {
     ipsum = ip_checksum(tvb_get_ptr(tvb, offset, hlen), hlen);
     if (tree) {
       if (ipsum == 0) {
